lib/IDE/Formatting.cpp

```diff
@@ -0,0 +1,859 @@
+//===--- Formatting.cpp ---------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/IDE/Formatting.h"
+
+namespace swift {
+namespace ide {
```

Comment:
A lot of the code in here isn't actually part of the `swift::ide::` API.  Anything not part of the API should be inside a `namespace {` if it's a type, or marked `static` for global functions.

Also, please switch to `using namespace swift` `using namespace ide` in the implementation file instead of wrapping everything in namespaces.  Then the particular functions you want to expose can be declared qualified like `swift::ide::reformat(...`.  See ModuleInterfacePrinting.cpp for examples.